### PR TITLE
Add CentOS BareMetal deployment scripts.

### PR DESCRIPTION
**Pull request by kevin-wangzefeng**
Automated scripts to deploy kubernetes on CentOS bare-metal

ref #12914

### CODE REVIEW

**Comment by piosz** [comment]
@k8s-bot ok to test

**Comment by piosz** [comment]
@coolsvap, @nikhiljindal, @derekwaynecarr, @thockin 
You were involved in  [Manual Config Guide on CentOS](https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/centos/centos_manual_config.md). Could you please take a look?

**Comment by coolsvap** [comment]
I will test and verify it.

**Comment by nikhiljindal** [comment]
Thanks @coolsvap

**Comment by nikhiljindal** [comment]
@k8s-bot ok to test

**Comment by kevin-wangzefeng** [comment]
@coolsvap Hi, how is the test going on?

**Comment by nikhiljindal** [comment]
Assigning to @caesarxuchao since I am out for the next few days

**Comment by caesarxuchao** [comment]
@kevin-wangzefeng, may we postpone the review until @nikhiljindal comes back at the end of next week? I'm not familiar with this part of the code. Sorry for the delay.

**Comment by kevin-wangzefeng** [comment]
OK @caesarxuchao  
@nikhiljindal, could you please help to review this PR when you are back? Thanks!

**Comment by brendandburns** [comment]
LGTM.  thanks for the PR.

**Comment by dchen1107** [comment]
cc/ @lvlv since you mentioned that you could own BareMetal CentOS deployment scripts.

File: cluster/centos/.gitignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+binaries
+
+master/bin/etcd
+master/bin/etcdctl
+master/bin/kube*
+
+minion/bin/brctl
+minion/bin/docker
+minion/bin/etcd
```
Comment (lvlv) [inline cluster/centos/.gitignore]: as we are deprecating word "minion", can you rephrase minion to node?

File: cluster/centos/.gitignore [hunk 2]
```diff
@@ -0,0 +1,13 @@
+binaries
+
+master/bin/etcd
+master/bin/etcdctl
+master/bin/kube*
+
+minion/bin/brctl
```
Comment (lvlv) [inline cluster/centos/.gitignore]: I suppose this binary is strictly bundled with CentOS release version, can we install it in every machine instead of providing it here?

File: cluster/centos/config-build.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## Contains configuration values for the Binaries downloading and unpacking.
+
+# Directory to store release packages that will be downloaded.
+RELEASES_DIR=${RELEASES_DIR:-/tmp/downloads}
+
+# Define flannel version to use.
+FLANNEL_VERSION=${FLANNEL_VERSION:-"0.4.0"}
```
Comment (lvlv) [inline cluster/centos/config-build.sh]: can we use 0.5.3 or latest? 0.5.3 works for me, and we did fix some bugs affecting to our CentOS 7 deployment since 0.4.0 :-)
see: https://github.com/coreos/flannel/releases for details

File: cluster/centos/config-default.sh [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## Contains configuration values for the CentOS cluster
+
+# Currently only support root user.
+export MASTER=${MASTER:-"root@8.8.8.18"}
```
Comment (lvlv) [inline cluster/centos/config-default.sh]: 8.8.8.0/24 is a public IP range, which belongs to Google ... can you change it to 172.16.x.x by default?

File: cluster/centos/minion/bin/remove-docker0.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+# Delete default docker bridge, so that docker can start with flannel network.
+
+# exit on any error
+set -e
+
+rc=0
+ip link show docker0 >/dev/null 2>&1 || rc="$?"
+if [[ "$rc" -eq "0" ]]; then
+  ip link set dev docker0 down
+  /opt/kubernetes/bin/brctl delbr docker0
```
Comment (lvlv) [inline cluster/centos/minion/bin/remove-docker0.sh]: use "ip link delete docker0" directly here so you can remove the dependency on brctl

**Comment by kevin-wangzefeng** [comment]
@lvlv Thanks very much for your review, I will update the scripts and test.

**Comment by wojtek-t** [comment]
Jenkins timed out.

@k8s-bot test this please

File: cluster/centos/master/scripts/apiserver.sh [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+KUBE_SERVICE_ADDRESSES="--service-cluster-ip-range=${SERVICE_CLUSTER_IP_RANGE}"
+
+# --admission-control="AlwaysAdmit": Ordered list of plug-ins 
+# to do admission control of resources into cluster. 
+# Comma-delimited list of: 
+#   LimitRanger, AlwaysDeny, SecurityContextDeny, NamespaceExists, 
+#   NamespaceLifecycle, NamespaceAutoProvision, DenyExecOnPrivileged, 
+#   AlwaysAdmit, ServiceAccount, ResourceQuota
+#KUBE_ADMISSION_CONTROL="--admission-control=\"${ADMISSION_CONTROL}\""
+
```
Comment (dalanlan) [inline cluster/centos/master/scripts/apiserver.sh]: Quick question please. Why do you disable `KUBE_ADMISSION_CONTROL` here?

Comment (dalanlan) [inline cluster/centos/master/scripts/apiserver.sh]: @kevin-wangzefeng

Comment (dalanlan) [inline cluster/centos/master/scripts/apiserver.sh]: ping @brendandburns who reviewed this pr
Is there any particular reason for centos to disable `ADMISSION_CONTROL` when deploying?

Comment (kevin-wangzefeng) [inline cluster/centos/master/scripts/apiserver.sh]: @dalanlan There's no particular reason to disable, just not finished that time, I will send another PR to enable it.

Comment (dalanlan) [inline cluster/centos/master/scripts/apiserver.sh]: Please keep me in the loop if it doesn't bother you:)

Comment (kevin-wangzefeng) [inline cluster/centos/master/scripts/apiserver.sh]: OK
